app/controllers/users_controller.rb

```diff
@@ -21,15 +21,22 @@ def contact_info
   end
 
   def get_preferences
-    response = echo_client.get_preferences(get_user_id, token)
-    render json: response.body, status: response.status
+    preferences_response = echo_client.get_preferences(get_user_id, token, echo_client, session[:access_token])
+
+    urs_response = echo_client.get_urs_user(session[:user_name], session[:access_token])
```

Comment:
This also gets called if the user has visited Reverb, and will overwrite whatever preferences are stored in ECHO. I think that may be what users would expect, though, so I wouldn't change it.